src/Core/include/OpenLoco/Core/EnumFlags.hpp

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include <cassert>
+#include <cstdint>
+
+namespace OpenLoco::Core
+{
+
+#define OPENLOCO_ENABLE_ENUM_OPERATORS(ENM)                                                                                 \
+    static_assert(std::is_unsigned_v<std::underlying_type_t<ENM>> == true, "Underlying enum type must be unsignd");         \
```

Comment:
Do you have a suite of tests for this? Would be nice just as an example of how to use.

Reply:
I've included some tests now